### PR TITLE
[donotmerge] Prevent tearDownAfterClass exceptions from killing PHPUnit

### PR DESCRIPTION
**Pull request by epdenouden**
New version with a fixed rebase: https://github.com/sebastianbergmann/phpunit/pull/3490

### CODE REVIEW

**Comment by epdenouden** [comment]
Code coverage will be brought back by the refreshed tests I'm making for the basic end-to-end suite.

**Comment by epdenouden** [comment]
@sebastianbergmann I just discovered there is already a test for this exact case, which is run and green for `TestCast` but not `TestSuite`: https://github.com/sebastianbergmann/phpunit/blob/master/tests/_files/ExceptionInTearDownTest.php tested in:

https://github.com/sebastianbergmann/phpunit/blob/e8850c1dce9250e7b430a5d666feb7f71a5d658b/tests/unit/Framework/TestCaseTest.php#L155-L165

I'll fix this nicely and update this pull request. 🚜

File: tests/unit/Framework/TestSuiteTest.php [hunk 1]
```diff
@@ -227,4 +227,24 @@ public function testCreateTestForConstructorlessTestClass(): void
...
+     */
+    public function testTearDownAfterClassInTestSuite(): void
+    {
+        $suite = new TestSuite(\ExceptionInTearDownAfterClassTest::class);
+        $suite->run($this->result);
+
+        $this->assertSame(3, $this->result->count());
+        $this->assertCount(1, $this->result->failures());
+
+        /** @var TestFailure $failure */
```
Comment (sebastianbergmann) [inline tests/unit/Framework/TestSuiteTest.php]: Please use `\assert($failure instanceof TestFailure);` instead.

File: tests/_files/ExceptionInTearDownAfterClassTest.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+use PHPUnit\Framework\TestCase;
+
+class ExceptionInTearDownAfterClassTest extends TestCase
+{
+    public $setUp                = false;
```
Comment (sebastianbergmann) [inline tests/_files/ExceptionInTearDownAfterClassTest.php]: Is there a specific reason why these attributes are `public`?

File: tests/unit/Framework/TestSuiteTest.php (hunk 1 above)
Comment (sebastianbergmann) [reply tests/unit/Framework/TestSuiteTest.php]: As of b880c49bfec871a43702bca8c42b36ccc246f989 neither should be needed anymore.

File: tests/_files/ExceptionInTearDownAfterClassTest.php (hunk 2 above)
Comment (epdenouden) [reply tests/_files/ExceptionInTearDownAfterClassTest.php]: No anymore no, thanks for catching this. The test started off as a version of `tests/_files/ExceptionInTearDownTest.php`. I have cleaned up test to the minimum required.

File: tests/unit/Framework/TestSuiteTest.php (hunk 1 above)
Comment (epdenouden) [reply tests/unit/Framework/TestSuiteTest.php]: Will that commit also come to 7.5? I have added the `\assert()` for now.

**Comment by epdenouden** [comment]
I'm going to need to figure out a better way of juggling multiple working versions (7 vs 8) and branches of stuff I'm working on. Don't even let me command a nuclear sub. ONE REBASE ONLY.

**Pull request closed without merging**